ml/curatedBreastData-InfoGAN/funs_common.py

```diff
@@ -0,0 +1,120 @@
+import pandas as pd
+import os
+import numpy as np
+import random
+import math 
+from xgboost import XGBClassifier
+from sklearn.metrics import recall_score
+from sklearn.model_selection import RepeatedStratifiedKFold
+from funs_balance import random_upsample_balance
+from collections import defaultdict
+from sklearn import svm
+from sklearn.linear_model import LogisticRegression
+
+
+def convert_surgery(x): 
+    if (x == "mastectomy"): 
+        return 1 
+    if (x == "breast preserving"): 
+        return 2 
+    if (x == 'NA'):  
+        return 0 
+    raise Exception("bad surgery") 
+
+def convert_posOutcome(x):
+    if (x == '2'):
+        return 1
+    return int(x)
+
+def read_treat_dataset(path="./data"):
+    return pd.read_csv(os.path.join(path, 'bmc15mldata1.csv'), converters=dict(surgery = convert_surgery, posOutcome = convert_posOutcome))
+
+
+def read_coincide_types_dataset(path="./"):
+    dataset = pd.read_csv(os.path.join(path, 'coincideTypes.csv'))
+    treat_dataset = read_treat_dataset()
+    return pd.merge(treat_dataset, dataset)
+
+def read_alltreat_dataset(path="./"):
+    atreat_dataset = pd.read_csv(os.path.join(path, 'bcClinicalTable.csv'))
+    treat_dataset  = read_treat_dataset()
+    return pd.merge(treat_dataset, atreat_dataset)
+    
+
+def read_pam_types_cat_dataset():
+    coincide_types_dataset = read_coincide_types_dataset()
+    keep = coincide_types_dataset[["patient_ID", "study", 'pCR', 'RFS', 'DFS','radio','surgery','chemo','hormone', 'posOutcome']]
+    return pd.concat([keep, pd.get_dummies(coincide_types_dataset["pam_name"])], axis=1)
+        
+def read_full_dataset(path = "example15bmc"):
```

Comment:
there is ml/data_util directory for data loading. It makes sense to separate data loading code from neural network code since this code can be used for different models.

Reply:
Agree but I had to prepare and structurize data different way. Also functional approach is not a plus here because I have to keep track of many identifiers in my train/test scripts.

Reply:
this function seems to be unused